module/VuFindTheme/src/VuFindTheme/ResourceContainer.php

```diff
@@ -160,11 +184,46 @@ public function getCss()
     /**
      * Get Javascript files.
      *
+     * @param string $position Position where the files should be inserted
+     * (allowed values are 'header' or 'footer').
+     *
+     * @return array
+     */
+    public function getJs(string $position=null)
+    {
+        $this->js = array_unique($this->js, SORT_REGULAR);
+        if (!isset($position)) {
+            return $this->js;
+        } else {
+            $js = [];
```

Comment:
A functional approach might be slightly more efficient here:

<pre>
return array_filter($this->js, function ($jsFile) use ($position) {
    return $jsFile['position'] == $position;
});
</pre>